Geometry/HGCalCommonData/src/HGCalDDDConstants.cc

```diff
@@ -1519,55 +1581,63 @@ int HGCalDDDConstants::cellHex(
 }
 
 void HGCalDDDConstants::cellHex(
-    double xloc, double yloc, int cellType, int& cellU, int& cellV, bool extend, bool debug) const {
-  int N = (cellType == 0) ? hgpar_->nCellsFine_ : hgpar_->nCellsCoarse_;
-  double Rc = 2 * rmax_ / (3 * N);
-  double rc = 0.5 * Rc * sqrt3_;
-  double RcT = (extend) ? (2 * rmaxT_ / (3 * N)) : Rc;
-  double rcT = 0.5 * RcT * sqrt3_;
-  double v0 = ((xloc / Rc - 1.0) / 1.5);
-  int cv0 = (v0 > 0) ? (N + (int)(v0 + 0.5)) : (N - (int)(-v0 + 0.5));
-  double u0 = (0.5 * yloc / rc + 0.5 * cv0);
-  int cu0 = (u0 > 0) ? (N / 2 + (int)(u0 + 0.5)) : (N / 2 - (int)(-u0 + 0.5));
-  cu0 = std::max(0, std::min(cu0, 2 * N - 1));
-  cv0 = std::max(0, std::min(cv0, 2 * N - 1));
-  if (cv0 - cu0 >= N)
-    cv0 = cu0 + N - 1;
-#ifdef EDM_ML_DEBUG
-  if (debug)
-    edm::LogVerbatim("HGCalGeom") << "cellHex: input " << xloc << ":" << yloc << ":" << cellType << " parameter " << rc
-                                  << ":" << Rc << " u0 " << u0 << ":" << cu0 << " v0 " << v0 << ":" << cv0;
-#endif
-  bool found(false);
-  static constexpr int shift[3] = {0, 1, -1};
-  for (int i1 = 0; i1 < 3; ++i1) {
-    cellU = cu0 + shift[i1];
-    for (int i2 = 0; i2 < 3; ++i2) {
-      cellV = cv0 + shift[i2];
-      if (((cellV - cellU) < N) && ((cellU - cellV) <= N) && (cellU >= 0) && (cellV >= 0) && (cellU < 2 * N) &&
-          (cellV < 2 * N)) {
-        double xc = (1.5 * (cellV - N) + 1.0) * Rc;
-        double yc = (2 * cellU - cellV - N) * rc;
-        if ((std::abs(yloc - yc) <= rcT) && (std::abs(xloc - xc) <= RcT) &&
-            ((std::abs(xloc - xc) <= 0.5 * RcT) || (std::abs(yloc - yc) <= sqrt3_ * (RcT - std::abs(xloc - xc))))) {
-#ifdef EDM_ML_DEBUG
-          if (debug)
-            edm::LogVerbatim("HGCalGeom")
-                << "cellHex: local " << xc << ":" << yc << " difference " << std::abs(xloc - xc) << ":"
-                << std::abs(yloc - yc) << ":" << sqrt3_ * (Rc - std::abs(yloc - yc)) << " comparator " << rc << ":"
-                << Rc << " (u,v) = (" << cellU << "," << cellV << ")";
-#endif
-          found = true;
-          break;
+    double xloc, double yloc, int cellType, int place, int part, int& cellU, int& cellV, bool extend, bool debug) const {
+  if (mode_ == HGCalGeometryMode::Hexagon8Cassette) {
+    auto uv = (part == HGCalTypes::WaferFull)
+                  ? hgcellUV_->cellUVFromXY3(xloc, yloc, place, cellType, extend, debug)
+                  : hgcellUV_->cellUVFromXY1(xloc, yloc, place, cellType, part, extend, debug);
+    cellU = uv.first;
+    cellV = uv.second;
+  } else if (waferHexagon8File()) {
+    auto uv = hgcellUV_->cellUVFromXY3(xloc, yloc, place, cellType, extend, debug);
+    cellU = uv.first;
+    cellV = uv.second;
+  } else {
+    int N = (cellType == 0) ? hgpar_->nCellsFine_ : hgpar_->nCellsCoarse_;
+    double Rc = 2 * rmax_ / (3 * N);
+    double rc = 0.5 * Rc * sqrt3_;
+    double RcT = (extend) ? (2 * rmaxT_ / (3 * N)) : Rc;
```

Comment:
Variable names should start with lowercase.

Reply:
Trying. to take care of this